ValveKeyValue/ValveKeyValue.Test/Test Data/Text/conditional_between_key_and_value.vdf

```diff
@@ -0,0 +1,4 @@
+﻿"test case"
+{
+	"operating system" [$WIN32]		"windows 32-bit"
```

Comment:
Does Valve's KV actually support this?

What happens if you put a conditional before the key?

Reply:
I had a friend do some testing and it appears that the conditional can be before the key, between the key and value, or after the value:

![unknown](https://user-images.githubusercontent.com/873012/193352359-c8467580-c4a6-4169-a48e-742706f57758.png)
![unknown](https://user-images.githubusercontent.com/873012/193352379-1591665c-dd1e-44df-b722-6ba15e158dab.png)
![unknown](https://user-images.githubusercontent.com/873012/193352425-e1cc1206-104e-48c9-9293-2311efe5b29a.png)
![unknown](https://user-images.githubusercontent.com/873012/193352439-4ca76803-b75a-49bd-bb50-c5b97d60cc8e.png)

Reply:
I'm not sure that conditional before key is right. Are you sure it didn't apply to the previous key and it just used the first labelText (without the conditional)?

Reply:
Good catch, you are correct about conditional before key actually having applied to the previous line. I have updated the code and test to reflect this case as being invalid.

Reply:
I briefly looked at Valve's code, and I think they do explicitly support conditionals between key and value (on the root key, it's supported before the `{` which is a good test to add).

Confusingly they have an error: `g_KeyValuesErrorStack.ReportError("RecursiveLoadFromBuffer:  got conditional between key and value" );` which I don't know when actually gets triggered.

Reply:
This seems to be supported by accident, as that line does not get triggered with this sample input.

I suspect its a logic bug in the error condition. In [tier1/keyvalues.cpp](https://github.com/ValveSoftware/source-sdk-2013/blob/0d8dceea4310fde5706b3ce1c70609d72a38efdf/mp/src/tier1/KeyValues.cpp#L2465), Valve are relying on `wasConditional` which gets reset to `false` when reading the value after the condition. If I replace it with `bFoundConditional`, then the error fires appropriately.

Therefore I'm somewhat hesitant to support this case.

Reply:
This somewhat ties into the #28 issue.

Reply:
From what I can tell, supporting [conditionals before object values](https://github.com/SteamDatabase/ValveKeyValue/blob/3da5293b3225067924a51cc59ce161fa22b71ff0/ValveKeyValue/ValveKeyValue.Test/Test%20Data/Text/conditional_before_object_value.vdf), which is valid and appears in TF2 many times (and probably other games as well), means also supporting [conditionals between keys and non-object values](https://github.com/SteamDatabase/ValveKeyValue/blob/3da5293b3225067924a51cc59ce161fa22b71ff0/ValveKeyValue/ValveKeyValue.Test/Test%20Data/Text/conditional_between_key_and_value.vdf). Is there a way to differentiate between the two and only support the former?

Reply:
Probably not, however I am tempted to explicitly support the "before object values" case, and simply leave the "before non-object values" case as undefined behaviour (no unit test) - if it works then so be it, if it doesn't work then so be it, and if it breaks in future then so be it too.

Reply:
I think it's fine with tests.